auth/design.md

```diff
@@ -0,0 +1,68 @@
+﻿
+# etcd v3 authentication design
+
+## Why not using the existing v2 auth mechanism?
+The v3 protocol uses gRPC as its transport instead of RESTful interface like v2. This new protocol provides an opportunity to iterate on and improve the v2 design. For example, v3 auth has connection based authentication, rather than v2's slower per-request authentication. Additionally, v2 auth's semantics tend to be unwieldy in practice with respect to reasoning about consistency, which will be described in the next sections. For v3, there is a well-defined description and implementation of the authentication mechanism which fixes the deficiencies in the v2 auth system.
+
+### Functionality requirements
+1. per connection authentication, not per request

```

Comment:
s/per connection/Per connection

Should these be `*` instead of `1.` since the ordering of the requirements doesn't matter?